stdlib/test/builtin/test_string.mojo

```diff
@@ -1082,6 +1073,78 @@ def test_indexing():
     assert_equal(a[2], "c")
 
 
+def test_string_iter():
+    var vs = String("123")
+
+    # Borrow immutably
+    fn conc(vs: String) -> String:
+        var c = String("")
+        for v in vs:
+            c += v
+        return c
+
+    assert_equal(123, atol(conc(vs)))
+
+    concat = String("")
+    for v in vs.__reversed__():
+        concat += v
+    assert_equal(321, atol(concat))
+
+    # TODO
+    # for v in vs:
+    #     v.unsafe_ptr().store(0, "1")
+
+    # # Borrow immutably
+    # var concat: String
+    # for v in vs:
+    #     concat += v
+
+    # assert_equal(111, atol(concat))
+
+    var items = List[String](
+        "mojo🔥",
+        "السلام عليكم",
+        "Dobrý den",
+        "Hello",
+        "שָׁלוֹם",
+        "नमस्ते",
+        "こんにちは",
+        "안녕하세요",
+        "你好",
+        "Olá",
+        "Здравствуйте",
+    )
+    var rev = List[String](
+        "🔥ojom",
+        "مكيلع مالسلا",
+        "ned ýrboD",
+        "olleH",
+        "םֹולָׁש",
+        "ेत्समन",
+        "はちにんこ",
+        "요세하녕안",
+        "好你",
+        "álO",
+        "етйувтсвардЗ",
+    )
+    var utf8_lengths = List(5, 12, 9, 5, 7, 6, 5, 5, 2, 3, 12)
+    var item_idx = 0
+    for item in items:
+        var amnt = 0
+        var byte_idx = 0
+        for v in item[]:
+            var byte_len = len(v)
+            assert_equal(item[][byte_idx : byte_idx + byte_len], v)
+            byte_idx += byte_len
+            amnt += 1
```

Comment:
This is cool, but can we somehow test this directly at least on one example? E.g. comparing the lengths of and iterating through and  `"mojo🔥"` and `["m", "o", "j", "o", "🔥"]` or something like this.